### DISCUSSION TITLE
Dynamic inventory and questions

### DISCUSSION THREAD

**Discussion by r3nor** [pyinfra-dev/pyinfra#1093] [Q&A] [answered]
I am wondering how to build a dynamic inventory. I have a database of hosts, that gets updated, so my inventory must come from an API request.

Then, I need to be able to "limit" by hostname rather than IP, but the ssh connection must be done to an IP. What I have now is;

```
# inventory.py
import requests

response = requests.get('https://api.something.com/v1/servers', headers=headers)
data = response.json()

servers = []
for server in data["items"]:
    servers.append((server["ip"], {"hostname": server["hostname"]}))
```

I would like to limit by hostname, yet use `ip` to connect to that host. Is that possible to do? In ansible I can have the `ansible_host` variable.

**Comment by Fizzadar** [comment, marked as answer]
Hi @r3nor - this is indeed possible using the `ssh_hostname` data key, ie:

```
# inventory.py
import requests

response = requests.get('https://api.something.com/v1/servers', headers=headers)
data = response.json()

servers = []
for server in data["items"]:
    servers.append((server["hostname"], {"ssh_hostname": server["ip"]}))
```

[All the available SSH options are listed here](https://docs.pyinfra.com/en/3.x/connectors/ssh.html#available-data).
